### PR TITLE
Re-enable mhp-bench

### PR DESCRIPTION
**Pull request by haichangsi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -223,7 +223,7 @@ function(add_mhp_ctest_impl)
   if(DEFINED AMC_TIMEOUT)
     set_tests_properties(${AMC_TEST_NAME} PROPERTIES TIMEOUT ${AMC_TIMEOUT})
   else()
-    set_tests_properties(${AMC_TEST_NAME} PROPERTIES TIMEOUT 60)
+    set_tests_properties(${AMC_TEST_NAME} PROPERTIES TIMEOUT 600)
```
Comment (lslusarczyk) [inline CMakeLists.txt:226]: you've changed default timeout for all tests
please change timeout just for long tests (TIMEOUT option for add_mhp_test

File: benchmarks/gbench/common/distributed_vector.cpp [hunk 2]
```diff
@@ -163,19 +163,19 @@ void check_reduce(T actual) {
...
-  for (auto _ : state) {
-    for (std::size_t i = 0; i < default_repetitions; i++) {
-      stats.rep();
-      actual = xhp::reduce(src);
-    }
-  }
-  check_reduce(actual);
-}
-DR_BENCHMARK(Reduce_DR);
+// template <class... BackendT> static void Reduce_DR(benchmark::State &state) {
```
Comment (lslusarczyk) [inline benchmarks/gbench/common/distributed_vector.cpp:166]: do you need to comment out whole test? at least compilation may still be checked and we will have original git history

File: benchmarks/gbench/common/distributed_vector.cpp [hunk 3]
```diff
@@ -163,19 +163,19 @@ void check_reduce(T actual) {
...
+//   Stats stats(state, sizeof(T) * src.size(), 0);
+//   for (auto _ : state) {
+//     for (std::size_t i = 0; i < default_repetitions; i++) {
+//       stats.rep();
+//       actual = xhp::reduce(src);
+//     }
+//   }
+//   check_reduce(actual);
+// }
+// DR_BENCHMARK(Reduce_DR);
```
Comment (lslusarczyk) [inline benchmarks/gbench/common/distributed_vector.cpp:178]: please report jira issue (type "workstream" - we use it for QA-bugs) about this test hanging, then a comment explaining why it is disabled and put jira id in the comment

File: benchmarks/gbench/mhp/CMakeLists.txt [hunk 4]
```diff
@@ -18,10 +18,14 @@ add_executable(
   stencil_1d.cpp
   stencil_2d.cpp
   chunk.cpp
-  mdspan.cpp
+  # mdspan.cpp
   mpi.cpp)
 # cmake-format: on
 
+if(NOT ENABLE_SYCL)
```
Comment (lslusarczyk) [inline benchmarks/gbench/mhp/CMakeLists.txt:25]: add jira issue about this ("workstream") and write a comment here why `mdspan.cpp` is just in non-sycl config + write jira id in this comment

File: benchmarks/gbench/mhp/stencil_2d.cpp [hunk 5]
```diff
@@ -465,48 +465,50 @@ DR_BENCHMARK(Stencil2D_Reference);
...
+//       dr::mhp::halo(stencil_steps % 2 ? b : a).exchange();
+//       q.parallel_for(sycl::range(row_slice, cols - 2), op).wait();
+//       std::swap(in, out);
+//     }
+//     // fails on devcloud
+//     // checker.check(stencil_steps % 2 ? b : a);
+//   }
+// }
+
+// DR_BENCHMARK(Stencil2D_SegmentedSYCL_DR);
```
Comment (lslusarczyk) [inline benchmarks/gbench/mhp/stencil_2d.cpp:512]: comment just this line of code + report jira issue and put its id in the comment

**Review by lslusarczyk — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (haichangsi) [reply CMakeLists.txt:226]: as I told you in a private message this is just for testing purposes, I'll change it

**Review by lslusarczyk — APPROVED** [review]

**Comment by lslusarczyk** [comment]
All OK, thank you, merged.

**Pull request merged**